### ISSUE TITLE
`git-revision-date-localized` + `blog` plugin requires specific config

### ISSUE THREAD

**Issue by perpil** [squidfunk/mkdocs-material#5336]
### Context

I was adding a blog to my site which already was using the `git-revision-date-localized` plugin  with the following setup:

```
  - git-revision-date-localized:
      enable_creation_date: true
      type: timeago
```

When I attempted to run `mkdocs serve` after following the instructions, it errored with the following.

```
ERROR    -  [git-revision-date-localized-plugin] Unable to find a git directory and/or git is not installed. To ignore this error, set option 'fallback_to_build_date:
            true'
ERROR    -  Error reading page 'blog/archive/2023.md':
Traceback (most recent call last):
  File "/opt/homebrew/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/homebrew/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/mkdocs/__main__.py", line 234, in serve_command
    serve.serve(dev_addr=dev_addr, livereload=livereload, watch=watch, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/mkdocs/commands/serve.py", line 83, in serve
    builder(config)
  File "/opt/homebrew/lib/python3.10/site-packages/mkdocs/commands/serve.py", line 76, in builder
    build(config, live_server=live_server, dirty=dirty)
  File "/opt/homebrew/lib/python3.10/site-packages/mkdocs/commands/build.py", line 308, in build
    _populate_page(file.page, config, files, dirty)
  File "/opt/homebrew/lib/python3.10/site-packages/mkdocs/commands/build.py", line 177, in _populate_page
    page.markdown = config.plugins.run_event(
  File "/opt/homebrew/lib/python3.10/site-packages/mkdocs/plugins.py", line 520, in run_event
    result = method(item, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/mkdocs_git_revision_date_localized_plugin/plugin.py", line 200, in on_page_markdown
    last_revision_timestamp = self.util.get_git_commit_timestamp(
  File "/opt/homebrew/lib/python3.10/site-packages/mkdocs_git_revision_date_localized_plugin/util.py", line 139, in get_git_commit_timestamp
    raise err
  File "/opt/homebrew/lib/python3.10/site-packages/mkdocs_git_revision_date_localized_plugin/util.py", line 110, in get_git_commit_timestamp
    git = self._get_repo(realpath)
  File "/opt/homebrew/lib/python3.10/site-packages/mkdocs_git_revision_date_localized_plugin/util.py", line 40, in _get_repo
    self.repo_cache[path] = Repo(path, search_parent_directories=True).git
  File "/opt/homebrew/lib/python3.10/site-packages/git/repo/base.py", line 266, in __init__
    self.working_dir: Optional[PathLike] = self._working_tree_dir or self.common_dir
  File "/opt/homebrew/lib/python3.10/site-packages/git/repo/base.py", line 347, in common_dir
    raise InvalidGitRepositoryError()
git.exc.InvalidGitRepositoryError
```

### Bug description

I'm not sure this is a bug or just worthy of documenting since it is related to the blog creating archives and categories.

When using `git-revision-date-localized` and adding a `blog`, you need to set the `fallback_to_build_date: true` flag for `git-revision-date-localized` to build/serve.

Alternatively you can turn off `archives` and `categories` in the `blog` plugin.

### Related links

- [Setting up a Blog](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/)
- [Document Dates](https://squidfunk.github.io/mkdocs-material/setup/adding-a-git-repository/?h=local#document-dates)

### Reproduction

[example.zip](https://github.com/squidfunk/mkdocs-material/files/11173256/example.zip)

### Steps to reproduce

Run:

1. `mkdocs serve`

You may need to run these first:

```
git init
```

Then add a file and commit it to see the exact same behavior.

```
git add docs/index.md
git commit -m "Repro"
```

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Thanks for reporting! Looping in @timvink, as this is very likely happening because the archive and category pages are dynamically generated files that are registered with MKDocs (= added to the `Files` collection), but have no representation on the filesystem (actually they have, but in a temp folder). Tim, would you consider this a bug in the `git-revision-date-localized-plugin`, or should we mention the necessary configuration setting in our documentation? I expect more users to trip over it once the blog is part of the community edition.

**Comment by timvink** [comment]
Hi @squidfunk , Hi @perpil ! 

I've already encountered the issue with generated files causing errors, specifically with the `mkdocs-gen-files-plugin`.  (see https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/issues/101 for the full story).

Long story short, if the blog plugin could add an attribute `generated_by` to all generated pages, `git-revision-date-localized-plugin` will automatically fall back to the current date for those pages (because that's when they were generated). 

https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/blob/1faaf2d61ea54ac4133ee9d46213bf746a537436/mkdocs_git_revision_date_localized_plugin/plugin.py#L199-L202

**Comment by squidfunk** [comment]
Ah awesome, we can definitely add that! That's actually a really good idea to establish as a quasi-standard.

**Comment by squidfunk** [comment]
After upgrading the `mkdocs-git-revision-date-localized` plugin to [1.2.0](https://github.com/timvink/mkdocs-git-revision-date-localized-plugin/releases/tag/v1.2.0), and adding `generated_by` in [`fe72fa9d6`](https://github.com/squidfunk/mkdocs-material-insiders/commit/fe72fa9d66972d877a15875568d6437d93ba0d65), I can confirm that this works without further configuration now. Funnily I already tripped over this and forgot it, but fixed it with:

``` yaml
- git-revision-date-localized:
      enabled: !ENV [CI, false]
      enable_creation_date: true
      type: timeago
      exclude:
        - blog/archive/*
        - blog/category/*
```

This is no longer necessary.

**Comment by squidfunk** [comment]
Released as 9.1.7+insiders-4.32.6!
